### PR TITLE
Add rule react component name

### PR DESCRIPTION
**Pull request by quesabe**
Closes PLA-48.

### CODE REVIEW

File: src/react/component-name.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      }
+
+      if (!matches) {
+        return
+      }
+
+      const [, _componentName, functionName] = matches
+
+      if (_componentName !== functionName) {
+        fail(`Component's dir name must have first letter capitalized: ${path}`)
```
Comment (gvidon) [inline src/react/component-name.ts:37]: @quesabe is it a proper message?

File: src/react/component-name.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      } catch (error: any) {
+        // Any component's dir must have index.tsx within it. If index.tsx file was not found then it is not a component's dir
+        return
+      }
+
+      if (!matches) {
+        return
+      }
+
+      const [, _componentName, functionName] = matches
```
Comment (gvidon) [inline src/react/component-name.ts:34]: @quesabe why is the `_componentName` underlined?

**Review by gvidon — CHANGES_REQUESTED** [review]

File: src/react/component-name.ts (hunk 1 above)
Comment (quesabe) [reply src/react/component-name.ts:37]: My bad. Forgot to correct it after copy-pasting. 😳

File: src/react/component-name.ts (hunk 2 above)
Comment (quesabe) [reply src/react/component-name.ts:34]: It clashes with the function name (the exported rule). My attempts to invent something better led to long and verbose names, thus I ended up with. just an underscore.

**Pull request merged**